src/components/Button/Button.js

```diff
@@ -9,45 +44,11 @@ import classnames from 'classnames';
  * ```
  */
 
-class Button extends React.Component {
-  constructor(props, context) {
-    super(props, context);
-
-    this.handleClick = this.handleClick.bind(this);
-  }
-
-  handleClick(ev) {
-    if (this.props.onClick) {
-      this.props.onClick(ev, this.props.text);
-    }
-  }
-
-  render() {
-    const {
-      children,
-      className,
-      text, disabled,
-      style,
-      primary,
-      selected,
-      danger,
-      type,
-    } = this.props;
-
-    const cl = classnames('weave-button', { primary, selected, danger });
-    return (
-      <button
-        style={style}
-        disabled={disabled}
-        onClick={this.handleClick}
-        className={className || cl}
-        type={type}
-      >
-        {children || text}
-      </button>
-    );
-  }
-}
+const Button = ({ children, text, onClick, ...props }) => (
+  <StyledButton {...props} onClick={e => onClick(e, text)}>
```

Comment:
This will pass through invalid props to the `<button />` element. If I do `<Button selected />`, React will throw a warning about invalid props.

Reply:
styled-components filters out invalid props for us, but this is not ideal and can cause problems due to the large number of props they have to allow through. There is discussion here: https://github.com/styled-components/styled-components/issues/439 - passing props explicitly is something we could try. Will push a change in a moment.